### PR TITLE
Update Chorus components

### PR DESCRIPTION
**Pull request by wrigleyDan**
- Elasticsearch 8.5.2: Comes now with the ability to combine approximate knn with other features. Removed the LTR plugin as this is unused in Chorus as of now.
- Kibana 8.5.2
- Quepid: 6.14.0
- Keycloak: 18.0.2
- MySQL: 5.7.40

Added healthchecks docker-compose services

Added a new service that helps set up the user kibana_system in Elasticsearch due to changes in the now used major version 8.x

### CODE REVIEW

File: keycloak/wait-for-keycloak.sh [hunk 1]
```diff
@@ -2,5 +2,5 @@ DOT='\033[0;37m.\033[0m'
 ERROR='\033[0;31m[QUICKSTART] '
 RESET='\033[0m' # No Color
 
-while [[ "$(curl -s 'http://keycloak:9080/auth/realms/chorus' | jq '.realm')" != '"chorus"' ]]; do printf ${DOT}; sleep 5; done
+while [[ "$(curl -s 'http://keycloak:9080/realms/chorus' | jq '.realm')" != '"chorus"' ]]; do printf ${DOT}; sleep 5; done
```
Comment (epugh) [inline keycloak/wait-for-keycloak.sh:5]: Nice change....

File: docker-compose.yml [hunk 2]
```diff
@@ -156,18 +200,22 @@ services:
...
     environment:
-      KEYCLOAK_USER: admin
-      KEYCLOAK_PASSWORD: password
+      KC_HTTP_PORT: 9080
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: password
       DB_VENDOR: h2
     volumes:
-      - ./keycloak/realm-config:/opt/jboss/keycloak/realm-config
+      - ./keycloak/realm-config/chorus-realm.json:/opt/keycloak/data/import/chorus-realm.json:ro
```
Comment (epugh) [inline docker-compose.yml:218]: looks a lot nicer.....

File: docker-compose.yml [hunk 3]
```diff
@@ -156,18 +200,22 @@ services:
...
+      test: [ "CMD", "curl", "--fail", "http://localhost:8080/health" ]
     ports:
       - 9080:9080
-      - 9443:9443
     environment:
-      KEYCLOAK_USER: admin
-      KEYCLOAK_PASSWORD: password
+      KC_HTTP_PORT: 9080
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: password
```
Comment (epugh) [inline docker-compose.yml]: so...    I go back and forth on the use of the `.env`...   But if we are introducing it, then maybe the `KEYCLOAK_ADMIN_PASSWORD` should be in there?

File: docker-compose.yml [hunk 4]
```diff
@@ -156,18 +200,22 @@ services:
       - "9114:9114"
 
   keycloak:
-    image: quay.io/keycloak/keycloak:16.1.1
+    image: quay.io/keycloak/keycloak:18.0.2
     container_name: keycloak
-    command: ["-b", "0.0.0.0", "-Dkeycloak.migration.action=import", "-Dkeycloak.migration.provider=dir", "-Dkeycloak.migration.dir=/opt/jboss/keycloak/realm-config", "-Dkeycloak.migration.strategy=OVERWRITE_EXISTING", "-Djboss.socket.binding.port-offset=1000", "-Dkeycloak.profile.feature.upload_scripts=enabled"]
+    hostname: keycloak
+    command: [ "start-dev", "--import-realm" ]
+    restart: unless-stopped
```
Comment (epugh) [inline docker-compose.yml:207]: these restart's are awesome...  I never used them...   also love the nice pretty use of `command`

File: docker-compose.yml [hunk 5]
```diff
@@ -127,6 +167,10 @@ services:
     environment:
       - GF_SECURITY_ADMIN_PASSWORD=password
       - GF_USERS_ALLOW_SIGN_UP=true
+    healthcheck:
+      test: [ "CMD", "wget", "--tries=1", "--spider", "http://localhost:3000" ]
```
Comment (epugh) [inline docker-compose.yml:171]: nice

File: docker-compose.yml [hunk 6]
```diff
@@ -31,13 +52,20 @@ services:
       - es_data:/usr/share/elasticsearch/data
     ports:
       - "9200:9200"
+    healthcheck:
+      test: curl -s http://localhost:9200 >/dev/null || exit 1
```
Comment (epugh) [inline docker-compose.yml]: maybe nitpicky, but is there a reason to use curl here and wget elsewhere?   and that this one has a /dev/null?

File: docker-compose.yml [hunk 7]
```diff
@@ -1,9 +1,28 @@
-version: '3.7'
+version: '2.2'
 services:
 
+  setup:
```
Comment (epugh) [inline docker-compose.yml:4]: this is interesting.....      I wonder though, the pattern in chorus has been that allof this kind of setup/validation logic, plus the "timing" logic, has all been in the `quickstart.sh`...   is it simpler this way, or more confusing to an end user?

Or is it that this logic here doesn't matter, so shouldn't be in quickstart...   that it's just plubming...   humm...

**Review by epugh — COMMENTED** [review]
Some great stuff here!  I am going back and forth on the use of the `.env`...   Would having a .env help simplify the weird logic in the solr chorus when we rename crap everywhere for the online version?

File: docker-compose.yml (hunk 6 above)
Comment (wrigleyDan) [reply docker-compose.yml]: Excellent point, I don't see any reason not to use the same format as for the other healthcases to have that more consistent. Changing that in the next commit.

File: docker-compose.yml (hunk 7 above)
Comment (wrigleyDan) [reply docker-compose.yml:4]: I've taken that directly from Elastic's documentation and adapted it to our use case. The only purpose of this is to set a password for a user (`kibana_system`) which was not necessary prior to Elasticsearch 8.x.
I guess we could add it to the `quickstart.sh` script but it really is just setting up a password for a system user that is only used within Kibana. As a user I don't really want to be bothered with these kinds of things and having them set automatically (without any kind of scripting) would be personal preference. So my argument is that it is easier if we have it in the compose file as it decreasess the steps that need to be taken to set everything up.

File: keycloak/wait-for-keycloak.sh (hunk 1 above)
Comment (wrigleyDan) [reply keycloak/wait-for-keycloak.sh:5]: We already use that in the Solr version of Chorus which was necessary to make the new Keycloak version work.

File: docker-compose.yml (hunk 2 above)
Comment (wrigleyDan) [reply docker-compose.yml:218]: Another change that I've taken from the Solr version of Chorus.

File: docker-compose.yml (hunk 4 above)
Comment (wrigleyDan) [reply docker-compose.yml:207]: I think I'm using your Solr-Chorus work here again ;)

File: docker-compose.yml (hunk 3 above)
Comment (wrigleyDan) [reply docker-compose.yml]: Missed that comment, you're right, we could and probably should put other credentials in the `.env` file. And all the other credentials that we set. And we should use variables as much as possible to minimize the effort to update these things. Next commit incoming...

File: docker-compose.yml (hunk 5 above)
Comment (wrigleyDan) [reply docker-compose.yml:171]: Again, just using your Solr Chorus work :)

File: docker-compose.yml (hunk 7 above)
Comment (epugh) [reply docker-compose.yml:4]: Makes sense...     Plus if that is how they recommend it.....

File: docker-compose.yml (hunk 3 above)
Comment (epugh) [reply docker-compose.yml]: i think variables for things that we really expect to change...    I don't mind hard coding things to make it as simple to understand as possible....    As we frequently say, you aren't expected to deploy this to PROD ;-)

Comment (epugh) [reply docker-compose.yml]: Though maybe in the future we rethink that???!!!!

**Pull request merged**